src/sdk/namespace_fs.js

```diff
@@ -1151,42 +1155,43 @@ class NamespaceFS {
                         latest_ver_info, open_mode, undefined, versioned_info) :
                         undefined;
 
-                dbg.log1('Namespace_fs._move_to_dest_ver_enabled:', latest_ver_info, new_ver_info, gpfs_options);
+                dbg.log1('Namespace_fs._move_to_dest_version:', latest_ver_info, new_ver_info, gpfs_options);
 
                 if (latest_ver_info) {
-                    dbg.log1('NamespaceFS._move_to_dest_ver_enabled latest version exist - moving existing latest version to .versions/');
-                    await this._make_path_dirs(versioned_path, fs_context);
-                    await this.safe_move(fs_context, latest_ver_path, versioned_path, latest_ver_info,
-                        gpfs_options && gpfs_options.move_to_versions);
+                    dbg.log1('NamespaceFS._move_to_dest_version: latest version exist');
+                    if (is_suspended_mode && (latest_ver_info.version_id_str === NULL_VERSION_ID)) {
+                        dbg.log1('NamespaceFS._move_to_dest_version: version ID of the latest version is null - the file will be unlinked');
+                        await this.safe_unlink(fs_context, latest_ver_path, latest_ver_info, gpfs_options);
+                    } else {
+                        dbg.log1('NamespaceFS._move_to_dest_version version ID of the latest version is a unique ID - the file will be moved it to .versions/ directory');
+                        await this._make_path_dirs(versioned_path, fs_context);
+                        await this.safe_move(fs_context, latest_ver_path, versioned_path, latest_ver_info,
+                            gpfs_options && gpfs_options.move_to_versions);
+                    }
                 }
+                // remove a version (or delete marker) with null version ID from .versions/ (if exists)
```

Comment:
any reason for this structure?
If possible I would prefer 
if (versioning_enabled) {
...
} else if (versioning_suspended){
if (latest is null) unlink latest
else delete_null_version_from_versions_dir()
} else {
throw err
}